### PR TITLE
Flatten book layout

### LINKED ISSUE #160: Flatten book layout structure
Rather than using `page-title/_index.md`, it would be very convenient to just be able to use `page-title.md`.

Unfortunately, I couldn't figure out how to get Zola to do this.

### PR DESCRIPTION
**Pull request by TrialDragon**
Makes the book use pages where before sections with just an _index.md file were for developer usability sake. Also does a minor change to the migration guides urls because apparently dots can't be in page urls turning it from `0.4-0.5` to `0-4-0-5`. Now they're like `0-4-to-0-5` for readability sake.

Closes #160

### CODE REVIEW

**Comment by TrialDragon** [comment]
Still need to fix next / previous page navigation.

**Comment by TrialDragon** [comment]
I'd like to know opinions on the whole migration guide's url thing. Should those just be reverted to sections to avoid that issue at the cost of losing minor usability benefits moving to pages bring, or is it worth it to just keep them as pages and deal with the url change? (I already tried change the slug and such; pages unlike sections just can't have dots in the page url apparently.)

**Comment by TrialDragon** [comment]
(This comment is more of a note to myself to look into this) This PR may require changing the migration guide stub generator since it may be generating the files as sections still (presuming migration guide stuff isn't switched back to sections.)

**Comment by alice-i-cecile** [comment]
> I'd like to know opinions on the whole migration guide's url thing. Should those just be reverted to sections to avoid that issue at the cost of losing minor usability benefits moving to pages bring, or is it worth it to just keep them as pages and deal with the url change? (I already tried change the slug and such; pages unlike sections just can't have dots in the page url apparently.)

I'd like to keep this consistent across the project and swap to pages. We should set up redirects though.

**Comment by TrialDragon** [comment]
> I'd like to keep this consistent across the project and swap to pages. We should set up redirects though.

By redirects you mean setup dummy section pages that go to regular pages since the sections can have dots in url?

**Comment by TrialDragon** [comment]
Ope, ignore the previous comment; aliases just work for redirects apparently? How did I not notice this when I was looking for solutions before... I'll commit that ASAP and get the conflicts solved.

**Review by alice-i-cecile — APPROVED** [review]

File: content/learn/book/contributing/_index.md
Comment (cart) [inline content/learn/book/contributing/_index.md:10]: Why is duplicating the "weight" field necessary? Remembering to add it twice is extra work that needs to be done, and it creates potential for the weight to be out of sync.
This one is out of sync with the "actual weight". Was that a mistake or intentional?

File: content/learn/book/next-steps.md
Comment (cart) [inline content/learn/book/next-steps.md:7]: This is out of sync with the "actual weight" (see my other comment about this).

**Review by cart — COMMENTED** [review]
Love the flattened structure. Just one thought / concern.

File: content/learn/book/contributing/_index.md
Comment (TrialDragon) [reply content/learn/book/contributing/_index.md:10]: In the docs menu macro we can't access the actual weight thru the zola pages / sections objects. It isn't a field. But I needed to resort them by weight due to the fact there isn't a mixed pages and sections array and I had to make one myself.

The out of sync wasn't intended.

Tbh, I'm 99% sure we can just replace zola's weight with the custom weights and be fine. I just left them in on the off chance I missed one esoteric thing that does need weights and the code I added doesn't replace in effective functionality and as such breaks things unexpectedly.

**Review by pablo-lua — COMMENTED** [review]
Looks really interesting changes in general, really flattened the book.

File: content/learn/book/contributing/_index.md
Comment (TrialDragon) [reply content/learn/book/contributing/_index.md]: Okay, have removed the duplication.

**Comment by alice-i-cecile** [comment]
@BD103, could you take a look at this PR? I'd like to get this merged in before we start working on the book content further.

**Comment by BD103** [comment]
> @BD103, could you take a look at this PR? I'd like to get this merged in before we start working on the book content further.

Yeah, I'll give it my best shot! :D

File: content/learn/advanced-examples/sudoku/_index.md
Comment (BD103) [inline content/learn/advanced-examples/sudoku/_index.md:4]: I may be wrong, but isn't `page_template` inherited? Since it is set in `advanced-examples/_index.md`, does it also need to be set here?

**Review by BD103 — COMMENTED** [review]
From a very brief first pass, there don't appear to be any visible changes from browsing the built website. (Which is good, since messing up URLs would be bad.)

My only question is about setting the template for every single page / section, when the docs mentions that it might be inheritable? Could this be reduced so that there is less copy-and-pasting when creating new pages?

File: content/learn/advanced-examples/sudoku/_index.md
Comment (TrialDragon) [reply content/learn/advanced-examples/sudoku/_index.md:4]: Honestly not sure. I'll do some tests later to check. Hmm.

File: content/learn/quick-start/_index.md
Comment (BD103) [inline content/learn/quick-start/_index.md:1]: Does this need an `extra.weight` property, even though it is empty?

File: content/learn/quick-start/next-steps.md
Comment (BD103) [inline content/learn/quick-start/next-steps.md:6]: This weight was changed from 12 to 3. Was that purposeful?

**Review by BD103 — APPROVED** [review]
I think this looks good! (Though there were a _lot_ of changed files, so I may have missed something.) I noticed a few things and wanted to make sure they were intentional, so check out my comments.

File: content/learn/book/plugin-development/_index.md
Comment (TrialDragon) [reply content/learn/book/plugin-development/_index.md:1]: Yes. That's an issue from the new book merge. That file was never supposed to be left, and I noticed it during this PR. I can, and probably should, spin this out into its own issue and PR tho.

Comment (BD103) [reply content/learn/book/plugin-development/_index.md:1]: Ok. If the file was never meant to be in the book in the first place, then you should probably make a separate PR deleting it. At the very least, that PR will likely get merged faster than this one.

Comment (TrialDragon) [reply content/learn/book/plugin-development/_index.md:1]: Did my best to restore it; will open issue and PR to remove it.

File: content/learn/quick-start/next-steps.md
Comment (TrialDragon) [reply content/learn/quick-start/next-steps.md:6]: Yes and no. It was originally 4, then made 3 on main during some re-ordering with plugin guidelines, the fact that it is 12 on main means the original new book to main merge fucked up some weights and wasn't caught. It's still supposed to be 3.

Hmm. Either time to open up an issue and fix that on main, or just wait until this is merged to avoid having to solve _another_ merge conflict.

File: content/learn/advanced-examples/sudoku/_index.md
Comment (TrialDragon) [reply content/learn/advanced-examples/sudoku/_index.md:4]: Ope, yep. I'll remove all the unnecessary `page_template`'s across the content then.

File: content/learn/quick-start/_index.md
Comment (TrialDragon) [reply content/learn/quick-start/_index.md:1]: I don't think so. The way stuff works it gets all of `learn/{current_section`'s subsections those are the ones with extra.weight called on them. If we ever, for whatever reason, have `quick-start` as a child of a `docs-section` (for example converting the learn page to one) then yes, it would need a weight, but that can be handled then since the site will refuse to compile at that point until solved.

Comment (BD103) [reply content/learn/quick-start/_index.md:1]: Ok, just making sure!

File: content/learn/quick-start/next-steps.md
Comment (BD103) [reply content/learn/quick-start/next-steps.md:6]: I don't think this is a blocking issue, so it can probably be merged as-is.

**Comment by alice-i-cecile** [comment]
Thanks a ton! This is much nicer to work with :)

**Pull request merged**
